src/components/form-editor/form-editor.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { Column, Row, Tabs, Tab } from "carbon-components-react";
+import { useTranslation } from "react-i18next";
+import { showToast } from "@openmrs/esm-framework";
+import SchemaEditorComponent from "./schema-editor/schema-editor.component";
+import styles from "./form-editor.scss";
+import { useParams } from "react-router-dom";
+import { useFormSchema, useFormMetadata } from "../../api/fetchFormDetail";
+import { Form, Schema } from "../../api/types";
+
+const FormEditor: React.FC = () => {
+  type State = {
+    uuid: string;
+  };
+  const { t } = useTranslation();
+  const { uuid } = useParams<State>();
+
+  const GetFormMetaData = (formUUID: string) => {
+    const { formMetaData, formMetaDataError } = useFormMetadata(formUUID);
+    if (formMetaDataError) {
+      showToast({
+        title: t("error", "Error"),
+        kind: "error",
+        critical: true,
+        description: `${formMetaDataError?.message}`,
+      });
+    }
+    return formMetaData;
+  };
+
+  const GetFormSchema = (form: Form) => {
+    const { formSchemaData, isSchemaLoading, formSchemaError } =
+      useFormSchema(form);
+    if (form?.resources.length !== 0) {
+      if (formSchemaError) {
+        showToast({
+          title: t("error", "Error"),
+          kind: "error",
+          critical: true,
+          description: `${formSchemaError?.message}`,
+        });
+      }
+      if (isSchemaLoading) {
+        return "Loading...";
+      }
+      return formSchemaData;
+    } else {
+      return GetRawSchema();
+    }
+  };
+
+  const GetRawSchema = () => {
+    const schema: Schema = {
+      name: "",
+      pages: [],
+      processor: "EncounterFormProcessor",
+      uuid: "xxx",
+      referencedForms: [],
+    };
+    return schema;
```

Comment:
This abstraction is unnecessary. Just declare this as a constant. I would declare this at the top level of this module (I mean out of the `FormEditor` component scope).

```javascript
import { Form, Schema } from "../../api/types";

const emptyFormSchema = {
      name: "",
      pages: [],
      processor: "EncounterFormProcessor",
      uuid: "xxx",
      referencedForms: [],
    };

const FormEditor: React.FC = () => { 
```